### PR TITLE
#606 Feat/test suite migration

### PR DESCRIPTION
**Pull request by tdadadavid**

### CODE REVIEW

**Review by apoorvyadav1111 — APPROVED** [review]
Tremendous Effort! looks good to me

**Comment by apoorvyadav1111** [comment]
Hi @tdadadavid, while the check is running, could you please create an issue and link this pr to it? I will merge this soon so that people can use new package in migration.

Thanks

**Comment by tdadadavid** [comment]
There is already an issue for this PR 
This is it #606

File: integration_tests/commands/async/json_test.go
Comment (apoorvyadav1111) [inline integration_tests/commands/async/json_test.go:1097]: This might be failing test cases

Comment (tdadadavid) [reply integration_tests/commands/async/json_test.go:1097]: Hmm, okay I will take a look at it now

**Comment by AshwinKul28** [comment]
Commendable efforts @tdadadavid thanks! 🥳

File: integration_tests/commands/async/json_test.go
Comment (tdadadavid) [reply integration_tests/commands/async/json_test.go:1097]: Running the test in isolation, it works, but running with others makes it fail.

json_test_fail


I also noticed all tests in ZCOUNT passes but at the end it shows FAIL
zcount_fail


Any thoughts on this?

Comment (apoorvyadav1111) [reply integration_tests/commands/async/json_test.go:1097]: The Fail could be from different test case that ran earlier. Can you please search for Fail and see if there are logs for this

Comment (c-harish) [reply integration_tests/commands/async/json_test.go:1097]: FAIL below the ZCOUNT is also from json_test.  
JSON.OBJKEYS with empty path TC in websocket is failing.

Comment (apoorvyadav1111) [reply integration_tests/commands/async/json_test.go:1097]: Do we have a fix for this one @c-harish ?

Comment (c-harish) [reply integration_tests/commands/async/json_test.go:1097]: @apoorvyadav1111. Assert failing TC with arraysArePermutations instead of equal. JSON.OBJKEYS orders are not deterministic.

Comment (tdadadavid) [reply integration_tests/commands/async/json_test.go:1097]: Okay I will do this now. Thanks

Comment (c-harish) [reply integration_tests/commands/async/json_test.go:1097]: Hi, @tdadadavid. Please user ElementsMatch for assertion of all json.objkeys testcases before merging. Thanks.

Comment (apoorvyadav1111) [reply integration_tests/commands/async/json_test.go:1097]: Thanks @c-harish, @tdadadavid You can also use elementsMatch https://pkg.go.dev/github.com/stretchr/testify/assert#ElementsMatch

**Review by apoorvyadav1111 — CHANGES_REQUESTED** [review]
Blocking merge until test cases with array as expected are fixed

**Review by apoorvyadav1111 — APPROVED** [review]

**Pull request merged**
